### PR TITLE
Always attempt to remove finalizer on delete

### PR DESCRIPTION
**Pull request by abays**
We need to make sure that the finalizer is removed from the CinderAPI CR regardless of what the current state of Cinder is within Keystone

### CODE REVIEW

File: controllers/cinderapi_controller.go [hunk 1]
```diff
@@ -256,18 +256,20 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
...
+			if err != nil && !k8s_errors.IsNotFound(err) {
 				return ctrl.Result{}, err
 			}
 
-			ksSvcObj := keystone.NewKeystoneService(ks.Spec, instance.Namespace, map[string]string{}, 10)
-			err = ksSvcObj.Delete(ctx, helper)
+			// Could get here if there was an error and it was "not found", so need to check
+			// (we do it this way because we want to make sure the "RemoveFinalizer" call is
+			// executed even if the KeystoneService no longer exists)
+			if err == nil && ks != nil {
```
Comment (stuggi) [inline controllers/cinderapi_controller.go:266]: we'd have to also change https://github.com/openstack-k8s-operators/keystone-operator/blob/master/pkg/external/funcs.go#L247 to return nil for the ks if not found.

Comment (abays) [reply controllers/cinderapi_controller.go:266]: Ah yeah, forgot about that.  I'll do that in another PR now.

Comment (abays) [reply controllers/cinderapi_controller.go:266]: Actually, for the moment, let's just check here for `ks != nil` only.  The required changes to the Keystone operator have ripple effects that I want to make sure I understand.

File: controllers/cinderapi_controller.go [hunk 2]
```diff
@@ -256,18 +256,20 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
...
+			if err != nil && !k8s_errors.IsNotFound(err) {
 				return ctrl.Result{}, err
 			}
 
-			ksSvcObj := keystone.NewKeystoneService(ks.Spec, instance.Namespace, map[string]string{}, 10)
-			err = ksSvcObj.Delete(ctx, helper)
+			// Could get here if there was an error and it was "not found", so need to check again
+			// (we do it this way because we want to make sure the "RemoveFinalizer" call is
+			// executed even if the KeystoneService no longer exists)
+			if !k8s_errors.IsNotFound(err) {
```
Comment (abays) [inline controllers/cinderapi_controller.go:266]: If `err` is `nil`, this returns `false`: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound

Comment (fmount) [reply controllers/cinderapi_controller.go:266]: Thanks @abays, I tried this patch and it works as expected in a local environment, so +1 on my side.
I'll do the same in [1] and test.

[1] https://github.com/openstack-k8s-operators/glance-operator/pull/34

**Review by fmount — APPROVED** [review]

File: controllers/cinderapi_controller.go (hunk 2 above)
Comment (abays) [reply controllers/cinderapi_controller.go:266]: Sounds good!

**Pull request merged**